src/behaviours/definition/index.js

```diff
@@ -0,0 +1,95 @@
+//Dependencies
+import React, {Component, PropTypes} from 'react';
+import {isNull, isUndefined, isArray, isString} from 'lodash/lang';
+
+// Import from focus-core
+// We need to investigate why import {getEntityInformations} from 'focus-core/entity/builder' didn't work, maybe an ES2015 related issue with babel.
+// Maybe because the node modules reads from the builded lib  instead of src.
+import {definition} from 'focus-core';
+const getEntityInformations =  definition.entity.builder.getEntityInformations;
+
+
+/**
+ * This function is a behaviour. It aims to comment a component to a definition.
+ *  - A definition is related to the data model
+ *  - Each field of the domain have a definition which contains its domain and the fact that it is required ot not.
+ *  - The definitions of your application should have been set using `focus-core/definition/entity/container/setEntityConfiguration`
+ * @param  {string | array} definitionPath - A string or an array of the definition path to the configuration.
+ * @param  {object} additionalDefinition - If you need to override a definition for a specific component, you can use this object.
+ * @return {function} - A function to commect a component to a definition.
+ * @example please read the end of the file.
+ */
+export default function definitionBehaviour(definitionPath, additionalDefinition){
+
+    // Arguments validation
+    if(isUndefined(definitionPath) || isNull(definitionPath)|| (!isString(definitionPath) && !isArray(definitionPath))){
+        throw new Error('the definition path should be givent in order to to know the domain of your entity property.');
+    }
+    if(!isUndefined(additionalDefinition) && !isNull(additionalDefinition) && !isObject(additionalDefinition)){
+        throw new Error('The additional definition if is defined should be an object');
+    }
+
+    // Definition Construction
+    const definitionConf = isArray(definitionPath) ? definitionPath : [definitionPath];
+    const definition = definitionConf.reduce((valeurPrecedente, valeurCourante) => ({...valeurPrecedente, ...getEntityInformations(definitionPath, additionalDefinition)}), {});
+
+    // annotation
+    // The wrapped component should have a props containing the definition object.
+    return function wrapComponentWithDefinition(ComponentToWrap) {
+
+        // Save the display name for later
+        const displayName = ComponentToWrap.displayName || 'Component';
```

Comment:
Are you sure this is working ? I had troubles trying to get this in another situation, I think I used the prototype if I remember correctly